mc/gui/breathing_widget.py

```diff
@@ -198,20 +198,19 @@ def start_breathing_in_timer(self):
 
     def add_from_dialog(self, i_ilist, i_olist):
         self.new_cycle_bool = True
-        counter_int = 0
-        while len(i_olist) > counter_int:
+        
+        for index, _ in enumerate(zip(i_ilist, i_olist)):
```

Comment:
This works, but i have an idea which may be even more elegant: What about using the second variable in the `index, _` tuple, which is itself a tuple (a tuple within a tuple)?

Something like this: `for index, (in_length_ft, out_length_ft) in enumerate(zip(i_ilist, i_olist))`

Not sure if this will work since i'm not sure if Python accepts doing variable unpacking of this more advanced kind but might be checking?

Kind Regards, Tord